packages/eslint-plugin-template/tests/rules/use-track-by-function.test.ts

```diff
@@ -0,0 +1,147 @@
+import {
+  convertAnnotatedSourceToFailureCase,
+  RuleTester,
+} from '@angular-eslint/utils';
+import rule, {
+  MessageIds,
+  RULE_NAME,
+} from '../../src/rules/use-track-by-function';
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+const ruleTester = new RuleTester({
+  parser: '@angular-eslint/template-parser',
+});
+
+const messageId: MessageIds = 'useTrackByFunction';
+
+ruleTester.run(RULE_NAME, rule, {
+  valid: [
+    `
+      <div *ngFor="let item of [1, 2, 3]; trackBy: trackByFn">
+        {{ item }}
+      </div>
+    `,
+    `
+      <div *ngFor="let item of [1, 2, 3]; let i = index; trackBy: trackByFn">
+        {{ item }}
+      </div>
+    `,
+    `
+      <div *ngFor="let item of [1, 2, 3]; trackBy : trackByFn">
+        {{ item }}
+      </div>
+    `,
+    `
+      <div *ngFor='let item of [1, 2, 3]; let i = index; trackBy: trackByFn'>
+        {{ item }}
+      </div>
+    `,
+    `
+      <div *ngFor  =  "let item of [1, 2, 3]; let i = index; trackBy : trackByFn">
+        {{ item }}
+      </div>
+    `,
+    `
+      <ng-template ngFor let-item [ngForOf]="[1, 2, 3]" let-i="index"
+        [ngForTrackBy]="trackByFn">
+        {{ item }}
+      </ng-template>
+    `,
+    `
+      <div *ngFor="let item of ['a', 'b', 'c']; index as i; trackBy: trackByFn">
+        {{ item }}
+      </div>
+
+      <ng-template ngFor let-item [ngForOf]="[1, 2, 3]" let-i="index"
+        [ngForTrackBy]="trackByFn">
+        {{ item }}
+      </ng-template>
+    `,
+    `
+      <div *ngFor="
+        let item of [1, 2, 3];
+        let i = index;
+        trackBy : trackByFn
+      ">
+    `,
+  ],
+  invalid: [
+    convertAnnotatedSourceToFailureCase({
+      description: 'should fail when trackBy function is not present',
+      annotatedSource: `
+        <ul>
+          <li *ngFor="let item of [1, 2, 3];">
+              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+            {{ item }}
+          </li>
+        </ul>
+      `,
+      messageId,
+    }),
+    //
+    // FIXME @angular/compiler can parse ngForTrackBy template text even if it's missing colon.
```

Comment:
Please could you remove this?

If there is work to be done in the angular compiler it should be tracked outside of the source code of this project